internal/upload/vmware/vmware.go

```diff
@@ -14,20 +14,36 @@ type Credentials struct {
 	Datacenter string
 	Cluster    string
 	Datastore  string
+	Folder     string
 }
 
-// ImportVmdk is a function that uploads a stream optimized vmdk image to vSphere
-// uploaded image will be present in a directory of the same name
-func ImportVmdk(creds Credentials, imagePath string) error {
+func commonOptions(creds Credentials) []string {
 	args := []string{
-		"import.vmdk",
 		fmt.Sprintf("-u=%s:%s@%s", creds.Username, creds.Password, creds.Host),
 		"-k=true",
 		fmt.Sprintf("-pool=%s/Resources", creds.Cluster),
 		fmt.Sprintf("-dc=%s", creds.Datacenter),
 		fmt.Sprintf("-ds=%s", creds.Datastore),
-		imagePath,
 	}
+	if creds.Folder != "" {
+		args = append(args, fmt.Sprintf("-folder=%s", creds.Folder))
+	}
+
+	return args
+}
+
+// ImportVmdk is a function that uploads a stream optimized vmdk image to vSphere
+// uploaded image will be present in a directory of the same name
+func ImportVmdk(creds Credentials, imagePath string) error {
+	args := []string{
+		"import.vmdk",
+	}
+	args = append(args, commonOptions(creds)...)
+	if creds.Folder != "" {
+		args = append(args, fmt.Sprintf("-folder=%s", creds.Folder))
+	}
+
+	args = append(args, imagePath)
```

Comment:
Is this appending the folder option twice?

Reply:
naw, `-folder=%s` is confusingly not a physical folder/path, it's just a logical folder which vsphere classifies the machine under.

Reply:
I mean, `commonOptions` does `args = append(args, ... creds.Folder)` and then after that it does the same, no?